synnax/pkg/distribution/ontology/retrieve.go

```diff
@@ -219,29 +218,36 @@ func (r Retrieve) retrieveEntities(
 	clause gorp.Retrieve[ID, Resource],
 	tx gorp.Tx,
 ) ([]Resource, error) {
-	entries := gorp.GetEntries[ID, Resource](clause.Params)
-	excludeFieldData := getExcludeFieldData(clause.Params)
-	includeSchema := getIncludeSchema(clause.Params)
-	for j, res := range entries.All() {
+	var (
+		entries          = gorp.GetEntries[ID, Resource](clause.Params)
+		excludeFieldData = getExcludeFieldData(clause.Params)
+		includeSchema    = getIncludeSchema(clause.Params)
+		retrieveResource = (!excludeFieldData) || includeSchema
+	)
+	// Iterate over the entries in place, retrieving the resource if the query requires it.
+	err := entries.MapInPlace(func(res Resource) (Resource, bool, error) {
```

Comment:
what does the bool represent in the return of this function passed in?

Reply:
Whether the returned item should be included in the mapped result. Essentially a built-in filter.

Reply:
Also documented in the function docstring :)